features/location/impl/src/main/kotlin/io/element/android/features/location/impl/AndroidLocationActions.kt

```diff
@@ -14,39 +14,48 @@
  * limitations under the License.
  */
 
-package io.element.android.features.location.impl.show
+package io.element.android.features.location.impl
 
 import android.content.Context
 import android.content.Intent
 import android.net.Uri
+import android.provider.Settings
 import androidx.annotation.VisibleForTesting
 import com.squareup.anvil.annotations.ContributesBinding
 import io.element.android.features.location.api.Location
+import io.element.android.features.location.impl.show.LocationActions
 import io.element.android.libraries.di.AppScope
 import io.element.android.libraries.di.ApplicationContext
 import timber.log.Timber
 import javax.inject.Inject
 
 @ContributesBinding(AppScope::class)
 class AndroidLocationActions @Inject constructor(
-    @ApplicationContext private val appContext: Context
+    @ApplicationContext private val context: Context
 ) : LocationActions {
-
-    private var activityContext: Context? = null
-
     override fun share(location: Location, label: String?) {
         runCatching {
             val uri = Uri.parse(buildUrl(location, label))
             val showMapsIntent = Intent(Intent.ACTION_VIEW).setData(uri)
             val chooserIntent = Intent.createChooser(showMapsIntent, null)
             chooserIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
-            appContext.startActivity(chooserIntent)
+            context.startActivity(chooserIntent)
         }.onSuccess {
             Timber.v("Open location succeed")
         }.onFailure {
             Timber.e(it, "Open location failed")
         }
     }
+
+    override fun openSettings() {
```

Comment:
There is already a `SystemUtils.openAppSettingsPage` function in the `:libraries:androidutils` module.

Reply:
Yes, I didn't use it because it expects an `activity` instance which we don't have here.

Reply:
I see it already does `addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)` , so technically we could use any Context there. Your call 🤷 .